source/game.database.bmx

```diff
@@ -2209,28 +2212,31 @@ Type TDatabaseLoader
 			'no production found
 			If Not earliestProductionData Then Continue
 
-			'person should be at least 5 years (fictional)
+			'(fictional) person should have a minimal age
 			Local dobYear:Int = GetWorldTime().GetYear( person.GetPersonalityData().GetDOB() )
 			Local ageOnProduction:Long = earliestProductionData.GetProductionStartTime() - person.GetPersonalityData().GetDOB()
 			Local ageOnProductionYears:Int = earliestProductionData.GetYear() - dobYear
-			Local adjustAge:Int = False
+			Local adjustAge:Int = -1
 
 			'in "time", not years - so babies are possible (eg. actors)
 			If ageOnProduction <= 0
+				'TODO this could still include director/script writer
 				TLogger.Log("TDatabase.FixLoadedData()", "Person ~q"+person.GetFullName()+"~q (GUID=~q"+person.getGUID()+"~q) is born in "+dobYear+". Impossible to have produced ~q"+earliestProductionData.GetTitle()+" in "+earliestProductionData.GetYear()+".", LOG_LOADING | LOG_WARNING)
-				adjustAge = True
-			ElseIf ageOnProductionYears < 5
+				adjustAge = 5
+			ElseIf ageOnProductionYears < minAgeForJobOrGenre
 				'too young director or scriptwriter
 				If earliestProductionData.HasCastPerson(person.GetID(), TVTPersonJob.DIRECTOR | TVTPersonJob.SCRIPTWRITER)
 					TLogger.Log("TDatabase.FixLoadedData()", "Person ~q"+person.GetFullName()+"~q (GUID=~q"+person.getGUID()+"~q) is born in "+dobYear+". Impossible to have produced ~q"+earliestProductionData.GetTitle()+" in "+earliestProductionData.GetYear()+". Directors and Scriptwriters need to be at least 5 years old.", LOG_LOADING | LOG_WARNING)
-					adjustAge = True
+					adjustAge = minAgeForJobOrGenre
+				Else
+					adjustAge = 5
```

Comment:
Koennten wir da hier (und an anderen Stellen) die 5 nicht mit einem `minAge` austauschen? und dann einfach nahe der `minAgeForJobOrGenre` definieren?

Reply:
Geht es nur darum, die Konstante rauszuziehen. Man bräuchte ja auf alle Fälle zwei Konstanten und allein ein Mindestalter reicht ja nicht, da das Mindestalter in unterschiedlichen Produktionsjahren unterschiedlich sein kann.
* 1985 Mindestalter 5 wegen "muss geboren sein
* 1990 Mindestalter 18 wegen "Adult"

Man könnte es natürlich so umschreiben, dass für jede Produktion das Mindest-Geburtsjahr ermittelt wird (im Prinzip dob-adjustAge) und am Ende, wenn das eigentliche Geburtsjahr später ist, das Geburtsjahr anpassen.